x-pack/elastic-agent/pkg/agent/cmd/container.go

```diff
@@ -532,26 +533,28 @@ func runLegacyAPMServer(streams *cli.IOStreams, path string) (*process.Info, err
 	if release.Snapshot() {
 		version = fmt.Sprintf("%s-SNAPSHOT", version)
 	}
-	// Extract the bundled apm-server binary into the APM_SERVER_PATH
+	// Extract the bundled apm-server into the APM_SERVER_PATH
 	if err := installer.Install(context.Background(), spec, version, path); err != nil {
 		return nil, errors.New(err,
 			fmt.Sprintf("installing %s (%s) from %s to %s", spec.Name, version, cfg.TargetDirectory, path))
 	}
-
-	// Start apm-server process respecting args
-	logInfo(streams, "Starting legacy apm-server daemon as a subprocess.")
-	pattern := filepath.Join(path, fmt.Sprintf("%s-%s-%s*", spec.Cmd, version, cfg.OS()), spec.Cmd)
-	files, err := filepath.Glob(pattern)
+	// Get the apm-server directory
+	files, err := ioutil.ReadDir(path)
 	if err != nil {
-		return nil, errors.New(err, fmt.Sprintf("searching apm-server in %s", pattern))
+		return nil, errors.New(err, fmt.Sprintf("reading directory %s", path))
 	}
-	if len(files) != 1 {
-		return nil, errors.New("multiple apm-server versions installed")
+	if len(files) != 1 || !files[0].IsDir() {
```

Comment:
What happens if `len(files) == 0`?

Reply:
good point, will do a follow up. I honestly wish there was some basic setup with ES, Fleet Server and Kibana, to run some integration tests and not only the e2e testing living in another repository.

Reply:
++ on the integration tests.

Reply:
https://github.com/elastic/beats/issues/23821#issuecomment-814080040

Reply:
> What happens if len(files) == 0?

`if len(files) != 1` covers the case that it is `0`, in which case the error is returned; 
@ruflin am I missing something here, did you have something else in mind?

Reply:
You are right, if it is 0 it only checks the first condition and also it never gets to line 549.